### PR TITLE
API documentation overhaul

### LINKED ISSUE #4022: Split off RECAP APIs to their own page
This follows #4011, but aims to do the same for the RECAP APIs:

-  Check all links to the documentation and update to new page, if applicable
-  Add test
-  Add to sitemap
-  Add to root help page

### PR DESCRIPTION
**Pull request by mlissner**
This is a big overhaul of our documentation. The main idea is that it splits our single-page documentation into distinct pages for:

 - Case Law
 - PACER APIs
 - RECAP APIs
 - Judges
 - Disclosures
 - Search
 - Citations
 - Visualizations
 - Alerts

This serves a number of goals
 - It gives us more space for things like examples and details.
 - It makes the pages more sharable and linkable. Rather than sharing and linking to a single huge page with an anchor, we (and users) can share a simple link.
 - It separates the main ideas (in the main documentation) from the API ideas (in the specific pages).
 - It makes the pages more google-able. Now a search for "PACER data API" should bring us right up.
 - It makes the main documentation page less intimidating.

On the flip side, this makes the documentation less CTRL+F-able. That's too bad.

Fixes #4022

### CODE REVIEW

**Comment by mlissner** [comment]
@flooie, this one is 95% HTML and 5% code, so it should be fairly quick to review. 

Do you mind taking a look, please, and merging if you're happy?

File: cl/alerts/templates/includes/old_alert_docket_list.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <ul>
   {% for docket in dockets %}
     <li><p style="font-weight: inherit; font-family: inherit; border: 0; vertical-align: baseline; font-style: inherit; margin: 0 0 1.5em; padding: 0;">
-      <a href="https://www.courtlistener.com{{ docket.get_absolute_url }}">
+      <a href="https://www.courtlistener.com{{ docket.get_absolute_url }}?order_by=desc">
```
Comment (mlissner) [inline cl/alerts/templates/includes/old_alert_docket_list.html:5]: This was to make these emails more actionable. Sorry, it's an unrelated change.

File: .editorconfig [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# http://editorconfig.org
+# https://editorconfig.org
```
Comment (mlissner) [inline .editorconfig:1]: Semgrep caught an http in my code, so I just fixed them across the code base.

File: cl/alerts/templates/alert_email_es.txt [hunk 3]
```diff
@@ -27,7 +27,7 @@ Disable this Alert (one click): https://www.courtlistener.com{% url "disable_ale
 ************************
 This alert brought to you by the 501(c)(3) non-profit Free Law Project
  - Blog: https://free.law
- - Twitter: http://twitter.com/freelawproject
+ - Twitter: https://twitter.com/freelawproject
```
Comment (flooie) [inline cl/alerts/templates/alert_email_es.txt:30]: are you ready to switch to x.com

File: cl/alerts/templates/docket_alert_email.txt [hunk 4]
```diff
@@ -45,7 +45,7 @@ To disable this alert unsubscribe here: https://www.courtlistener.com{% url 'tog
 This alert brought to you by the 501(c)(3) non-profit Free Law Project
 
  - Blog: https://free.law
- - Twitter: http://twitter.com/freelawproject
+ - Twitter: https://twitter.com/freelawproject
```
Comment (flooie) [inline cl/alerts/templates/docket_alert_email.txt:48]: see above.

File: cl/alerts/templates/docket_alert_unsubscription_email.txt [hunk 5]
```diff
@@ -11,7 +11,7 @@ https://www.courtlistener.com{% url 'toggle_docket_alert_confirmation' "subscrib
 This alert brought to you by the 501(c)(3) non-profit Free Law Project
 
  - Blog: https://free.law
- - Twitter: http://twitter.com/freelawproject
+ - Twitter: https://twitter.com/freelawproject
```
Comment (flooie) [inline cl/alerts/templates/docket_alert_unsubscription_email.txt:14]: see above.

File: cl/alerts/templates/emails/old_alert_email.txt [hunk 6]
```diff
@@ -42,7 +42,7 @@ This message brought to you by the 501(c)(3) non-profit Free Law Project
 
  - Blog: https://free.law/
 
- - Twitter: http://twitter.com/freelawproject/
+ - Twitter: https://twitter.com/freelawproject/
```
Comment (flooie) [inline cl/alerts/templates/emails/old_alert_email.txt:45]: see above.

File: cl/alerts/templates/pacer_alert_email.txt [hunk 7]
```diff
@@ -12,7 +12,7 @@ The PG&E bankruptcy appears to be available on PACER:
 --------
 This alert brought to you by the 501(c)(3) non-profit Free Law Project
  - Blog: https://free.law
- - Twitter: http://twitter.com/freelawproject
+ - Twitter: https://twitter.com/freelawproject
```
Comment (flooie) [inline cl/alerts/templates/pacer_alert_email.txt:15]: x

File: cl/api/templates/bulk-data.html [hunk 8]
```diff
@@ -136,6 +136,9 @@ <h3 id="opinions-db">Case Law Data</h3>
         </p>
       </li>
     </ul>
+    <p>We have also partnered with the Library Innovation Lab at Harvard Law Library to create <a href="https://huggingface.co/datasets/harvard-lil/cold-cases">a dataset on Hugging Face</a> with similar data.
```
Comment (flooie) [inline cl/api/templates/bulk-data.html:139]: I would say Harvard University Law Library but thats just me

File: cl/lib/pacer_session.py [hunk 9]
```diff
@@ -18,7 +18,7 @@ class ProxyPacerSession(PacerSession):
     PacerSession class to achieve the following:
 
     - Sets the 'X-WhSentry-TLS' header to 'true' for all requests.
-    - Replaces 'https://' with 'http://' in the URL before making the request.
+    - Replaces 'https://' with 'https://' in the URL before making the request.
```
Comment (flooie) [inline cl/lib/pacer_session.py:21]: this looks like a mistake ?

File: cl/alerts/templates/alert_email_es.txt (hunk 3 above)
Comment (mlissner) [reply cl/alerts/templates/alert_email_es.txt]: As you wish. I just pushed changes to remove Twitter from most places, and add BlueSky in a few. We need new icons, but that feels out of scope.

File: cl/alerts/templates/docket_alert_email.txt (hunk 4 above)
Comment (mlissner) [reply cl/alerts/templates/docket_alert_email.txt]: Done.

File: cl/alerts/templates/docket_alert_unsubscription_email.txt (hunk 5 above)
Comment (mlissner) [reply cl/alerts/templates/docket_alert_unsubscription_email.txt]: Done.

File: cl/alerts/templates/emails/old_alert_email.txt (hunk 6 above)
Comment (mlissner) [reply cl/alerts/templates/emails/old_alert_email.txt]: Done.

File: cl/alerts/templates/pacer_alert_email.txt (hunk 7 above)
Comment (mlissner) [reply cl/alerts/templates/pacer_alert_email.txt]: Done.

File: cl/api/templates/bulk-data.html (hunk 8 above)
Comment (mlissner) [reply cl/api/templates/bulk-data.html:139]: I wanted to get innovation in here.

File: cl/lib/pacer_session.py (hunk 9 above)
Comment (mlissner) [reply cl/lib/pacer_session.py]: You're right! Fixed.

File: cl/api/templates/bulk-data.html (hunk 8 above)
Comment (flooie) [reply cl/api/templates/bulk-data.html:139]: oh I meant after the innovation lab ...but its not an issue

**Comment by mlissner** [comment]
FYI, I just pushed a few more changes because I'm on a train and kind of bored. This:

 - Adds underlining in the navbar for the API docs
 - Redoes the navbar to include a bunch more links
 - Cleans up tabindex slightly (it's still bad) so that it can slowly get enhanced.

**Comment by flooie** [comment]
Great. Enjoy the train

File: cl/assets/templates/base.html [hunk 10]
```diff
@@ -246,40 +246,118 @@ <h1>You did not supply the "private" variable to your template.
...
+              <ul class="dropdown-menu" role="menu">
+                <li>
+                  <a href="{% url "advanced_fd" %}" tabindex="501">Search Financial Disclosures</a>
+                </li>
+                <li>
+                  <a href="{% url "coverage_fds" %}" tabindex="502">About this Collection</a>
+                </li>
+                <li>
+                  <a href="{% url "financial_disclosures_api_help" %}" tabindex="503">API</a>
+                </li>
```
Comment (flooie) [inline cl/assets/templates/base.html:352]: this is great but its causing tests to fail - 

I think you can insert 

```@timeout_decorator.timeout(SELENIUM_TIMEOUT)
def test_disclosure_homepage(self) -> None:
    """Can we load disclosure homepage?"""
    self.browser.get(self.live_server_url)
    dropdown = self.browser.find_element(By.ID, "navbar-fd")
    dropdown.click()
    link = self.browser.find_element(By.LINK_TEXT, "Search Financial Disclosures")
    link.click()
``` into this particular test

File: cl/api/templates/financial-disclosure-api-docs-vlatest.html [hunk 11]
```diff
@@ -64,6 +66,8 @@ <h1 id="about">Financial Disclosures&nbsp;API</h1>
   </p>
   <p>Judicial officers and certain judicial employees in the United States are required to file financial disclosure reports by <a href="https://www.law.cornell.edu/uscode/text/5a/compiledact-95-521/title-I">Title I of the Ethics in Government Act of 1978</a>. The Act requires that designated federal officials publicly disclose their personal financial interests to ensure confidence in the integrity of the federal government by demonstrating that they are able to carry out their duties without compromising the public trust.
   </p>
+  <p>These APIs were used by the Wall Street Journal in <a href="https://www.wsj.com/articles/131-federal-judges-broke-the-law-by-hearing-cases-where-they-had-a-financial-interest-11632834421">their 17-part exposé</a> about the hidden conflicts of federal judges. That led to Congress passing a law to put this information online. It was also used by ProPublica in <a href="https://www.propublica.org/article/clarence-thomas-scotus-undisclosed-luxury-travel-gifts-crow">their Pulitzer prize winning reporting</a> about failures to disclose gifts and perks.
```
Comment (flooie) [inline cl/api/templates/financial-disclosure-api-docs-vlatest.html:69]: how about this 

<p>These APIs were used by the Wall Street Journal in <a href="https://www.wsj.com/articles/131-federal-judges-broke-the-law-by-hearing-cases-where-they-had-a-financial-interest-11632834421">their 17-part exposé</a> about the hidden conflicts of federal judges. That led to Congress passing the <a href="https://www.govinfo.gov/content/pkg/PLAW-117publ125/pdf/PLAW-117publ125.pdf">Courthouse Ethics and Transparency Act</a> to put this information online. It was also used by ProPublica in <a href="https://www.propublica.org/article/clarence-thomas-scotus-undisclosed-luxury-travel-gifts-crow">their Pulitzer prize winning reporting</a> about failures to disclose gifts and perks.</p>

File: cl/api/templates/judge-api-docs-vlatest.html [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+        <li><a href="#retention-events">Retention Events</a></li>
+        <li><a href="#sources-endpoint">Sources</a></li>
+      </ul>
+    </ul>
+  </div>
+</div>
+
+
+<div class="col-xs-12 col-md-8 col-lg-6">
+  <h1 id="about">Judge and Justice&nbsp;API</h1>
```
Comment (flooie) [inline cl/api/templates/judge-api-docs-vlatest.html:52]: This narrowing of the database bothers me- but your mileage may vary.  

Could we switch to Judicial Directory API ?

File: cl/api/templates/judge-api-docs-vlatest.html [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+    </h4>
+    <h3>Table of Contents</h3>
+    <ul>
+      <li><a href="#about">Overview</a></li>
+      <li><a href="apis">The APIs</a></li>
+      <ul>
+        <li><a href="#people-endpoint">Judges &amp; Appointers</a></li>
+        <li><a href="#positions-endpoint">Positions Held</a></li>
+        <li><a href="#affiliations-endpoint">Political Affiliations</a></li>
+        <li><a href="#education-history">Educational Histories</a></li>
```
Comment (flooie) [inline cl/api/templates/judge-api-docs-vlatest.html:41]: Education Histories -> Educations and Schools ?

File: cl/api/templates/judge-api-docs-vlatest.html [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+    <li>
+      <p>If the <code>is_alias_of</code> field has a value, that means the record represents a nickname for the person referenced in the alias field. Alias records make it possible to find a judge by name, even if they sometimes go by Bob instead of Robert. In our database, this field is a <a href="https://en.wikipedia.org/wiki/Join_(SQL)#Self-join">self-join</a>.
+      </p>
+      <p>In general, you will only want to work with judges where this field is null, indicating a record that represents a person, not an alias to a person.
+      </p>
+    </li>
+    <li>
+      <p>The <code>race</code> and <code>gender</code> fields are not self-reported and should therefore be considered best guesses. We have done our best to gather these fields from reputable sources, but have also supplied values ourselves when it felt appropriate to do so. Some values may be incorrect.
+      </p>
+      <p>To our knowledge, there are no non-binary judges in the judiciary, making the categories for gender clear: Male or female.
```
[the remark this answers is not in the document]
Comment (flooie) [reply cl/api/templates/judge-api-docs-vlatest.html:108]: I would remove this and perhaps update our model in another PR

File: cl/api/urls.py [hunk 15]
```diff
@@ -179,30 +179,110 @@
...
+            template_name="visualizations-api-docs-vlatest.html",
+            extra_context={"private": False},
+        ),
+        name="visualization_api_help",
+    ),
     path(
         "help/api/rest/v3/financial-disclosures/",
-        views.financial_disclosures_api_help_help,
+        TemplateView.as_view(
+            # "financial-disclosure-docs-vlatest.html"
```
Comment (flooie) [inline cl/api/urls.py:241]: errant?

File: cl/api/urls.py [hunk 16]
```diff
@@ -179,30 +179,110 @@
...
+            template_name="alert-api-docs-vlatest.html",
+            extra_context={"private": False},
+        ),
         name="alert_api_help",
     ),
+    path(
+        "help/api/rest/v3/fields/",
+        TemplateView.as_view(
+            template_name="field-help.html",
+            extra_context={"private": True},
```
Comment (flooie) [inline cl/api/urls.py:267]: im not sure why this is set private=True

File: cl/assets/static-global/css/override.css [hunk 17]
```diff
@@ -891,7 +894,7 @@ input.court-checkbox, input.status-checkbox {
 
 /* Special case for Oregon, which has a name attr with a value of TOP on
    content that isn't a link. See example here:
-   http://www.publications.ojd.state.or.us/docs/S054822.htm#TOP
+   https://www.publications.ojd.state.or.us/docs/S054822.htm#TOP
```
Comment (flooie) [inline cl/assets/static-global/css/override.css:897]: this link ... doesnt work. so Im not sure what it means

File: cl/audio/templates/oral_argument.html [hunk 18]
```diff
@@ -203,7 +203,7 @@ <h3>{{ af.docket.court }}</h3>
                     </div>
                     <div class="jp-no-solution">
                         <span>Update Required</span>
-                        To play the media update your browser to a recent version or update your <a href="http://get.adobe.com/flashplayer/"
+                        To play the media update your browser to a recent version or update your <a href="https://get.adobe.com/flashplayer/"
```
Comment (flooie) [inline cl/audio/templates/oral_argument.html:206]: This is no longer accurate.  Flash has been sunset.  Thank goodness

File: cl/lib/context_processors.py [hunk 19]
```diff
@@ -35,7 +35,7 @@ def inject_settings(request):
     f'Free Law Project is a 501(c)(3) non-profit that relies on your support to operate. Please <a href="{DONATE_URL}?referrer=tip">donate</a> to support this site</a>. <i class="fa fa-heart red"></i>',
     f'CourtListener serves more than 200 million requests per year, but has a lean staff of only a handful of developers. Please <a href="{DONATE_URL}?referrer=tip">donate</a> to support our work making the judiciary more equitable. <i class="fa fa-heart red"></i>',
     f'CourtListener is supported by <a href="{DONATE_URL}?referrer=tip">user donations</a> and small grants. More donations result in less time spent seeking grants and more time adding features. <i class="fa fa-heart red"></i>',
-    f'Free Law Project is a member of the <a href="http://www.falm.info/">Free Access to Law Movement</a> and relies heavily on <a href="{DONATE_URL}?referrer=tip">your donations</a>. <i class="fa fa-heart red"></i>',
+    f'Free Law Project is a member of the <a href="https://www.falm.info/">Free Access to Law Movement</a> and relies heavily on <a href="{DONATE_URL}?referrer=tip">your donations</a>. <i class="fa fa-heart red"></i>',
```
Comment (flooie) [inline cl/lib/context_processors.py:38]: does not load https:// only http://

File: cl/visualizations/templates/visualization_home.html [hunk 20]
```diff
@@ -73,7 +73,7 @@ <h3 class="text-center">How it All Works</h3>
...
+                    In addition to showing the citation relationship between cases, this tool also leverages data about the cases available in the <a href="https://scdb.wustl.edu/index.php">Supreme Court Database (Spaeth)</a>. Specifically, the tool visualizes Spaeth data concerning the <a href="https://scdb.wustl.edu/documentation.php?var=decisionDirection">decision direction</a> of cases (liberal v. conservative) and <a href="https://scdb.wustl.edu/documentation.php?var=majVotes">vote count</a> (9-0, 8-1, etc). Links to the complete text of all opinions and links to the complete Spaeth data for a given case are also provided.
```
Comment (flooie) [inline cl/visualizations/templates/visualization_home.html:76]: no https for scdb.wustl.edu

**Comment by flooie** [comment]
This is much nicer and clearer across the board.  I would take a look though at the selenium tests that are failing now because of the tweaks and changes.  Otherwise, there are more http:// only websites out there than you might expect.

File: cl/api/templates/judge-api-docs-vlatest.html (hunk 12 above)
Comment (mlissner) [reply cl/api/templates/judge-api-docs-vlatest.html:52]: Hm, I actually find "directory" to be more narrow, no? It sort of implies that it's just a list of judges, as opposed to a database about them? I'd rather not think too hard about it for now though.

File: cl/api/templates/judge-api-docs-vlatest.html (hunk 13 above)
Comment (mlissner) [reply cl/api/templates/judge-api-docs-vlatest.html:41]: I sort of deliberately removed schools as an API that we talk much about because it's encompassed by the educational history semantics. For example, nobody wants to look up a particular school. What they want is the educational info of a judge.

File: cl/api/templates/judge-api-docs-vlatest.html (hunk 14 above)
[the remark this answers is not in the document]
Comment (mlissner) [reply cl/api/templates/judge-api-docs-vlatest.html:108]: I just removed the sentence. Reads fine without it.

File: cl/api/urls.py (hunk 15 above)
Comment (mlissner) [reply cl/api/urls.py:241]: yep, thanks.

File: cl/api/urls.py (hunk 16 above)
Comment (mlissner) [reply cl/api/urls.py:267]: I don't think anybody would really want to google this.

File: cl/assets/static-global/css/override.css (hunk 17 above)
Comment (mlissner) [reply cl/assets/static-global/css/override.css:897]: I just tweaked to https, so I don't think we need to solve this broken link.

File: cl/audio/templates/oral_argument.html (hunk 18 above)
Comment (mlissner) [reply cl/audio/templates/oral_argument.html:206]: Yeah, probably not, but we can solve it another day, out of scope for this (but I walked myself into this quandary by expanding my scope to all http links, doh!)

File: cl/lib/context_processors.py (hunk 19 above)
Comment (mlissner) [reply cl/lib/context_processors.py:38]: Lame. Good catch, thanks.

File: cl/api/templates/financial-disclosure-api-docs-vlatest.html (hunk 11 above)
Comment (mlissner) [reply cl/api/templates/financial-disclosure-api-docs-vlatest.html:69]: Nice. Done.

File: cl/simple_pages/templates/includes/available_fields.json [hunk 21]
```diff
@@ -25,7 +25,7 @@
    },
    {
       "field":"scdb_id",
-      "description":"The ID of an opinion in the <a href='http://scdb.wustl.edu/'>Supreme Court Database</a>.",
+      "description":"The ID of an opinion in the <a href='https://scdb.wustl.edu/'>Supreme Court Database</a>.",
```
[the remark this answers is not in the document]
Comment (mlissner) [reply cl/simple_pages/templates/includes/available_fields.json:28]: Lame. Fixed.

File: cl/visualizations/templates/visualization_home.html (hunk 20 above)
Comment (mlissner) [reply cl/visualizations/templates/visualization_home.html:76]: Got it.

**Comment by mlissner** [comment]
Thanks for offering to fix the test. I pushed your suggested change, but if it doesn't do the trick, I'd certainly welcome your help fixing that and pushing this over the line, and merging once it's all-clear.

I think I've addressed the rest of your suggestions (or decided not to). I'm proud you didn't find more typos, because, man, there was a lot to write!

**Comment by mlissner** [comment]
Yeah, OK, there are another five-or-so tests that need tweaking thanks to my navbar shenanigans. If you could smooth them out, I'd be quite grateful.

File: cl/api/templates/judge-api-docs-vlatest.html (hunk 13 above)
Comment (flooie) [reply cl/api/templates/judge-api-docs-vlatest.html:41]: thats interesting ... because I was using it to group judges by schools the other day, but I admit I may be an edge case

File: cl/audio/templates/oral_argument.html (hunk 18 above)
Comment (flooie) [reply cl/audio/templates/oral_argument.html:206]: I mean ... removing flash is always in scope to me... but I guess its a personal choice

File: cl/api/templates/judge-api-docs-vlatest.html (hunk 13 above)
Comment (mlissner) [reply cl/api/templates/judge-api-docs-vlatest.html:41]: Kind of the same thing, I think. Maybe an example of filtering to a particular school would help?

File: cl/audio/templates/oral_argument.html (hunk 18 above)
Comment (mlissner) [reply cl/audio/templates/oral_argument.html:206]: We'll get there. It's not causing any trouble in the meantime. When we get there, we can look at larger solutions to our janky audio player.

File: cl/api/templates/judge-api-docs-vlatest.html (hunk 13 above)
Comment (flooie) [reply cl/api/templates/judge-api-docs-vlatest.html:41]: I think that makes sense - let me add an example

File: cl/api/templates/judge-api-docs-vlatest.html (hunk 12 above)
Comment (flooie) [reply cl/api/templates/judge-api-docs-vlatest.html:52]: sure - im sure itll be fine
